main.tf

```diff
@@ -110,5 +110,5 @@ module "dns" {
   name    = module.this.name
   zone_id = var.zone_id
   ttl     = 60
-  records = aws_instance.default.*.public_dns
+  records = var.associate_public_address ? aws_instance.default.*.public_dns : aws_instance.default.*.private_dns
```

Comment:
variable name is associate_public_ip_address

Reply:
Ahh, stupid, used the online editor. I think it should be correct now.

I executed the commands below, but it failed on AWS creds.